src/Databases/PostgreSQL/FetchFromPostgreSQL.cpp

```diff
@@ -0,0 +1,118 @@
+#include <Databases/PostgreSQL/FetchFromPostgreSQL.h>
+
+#if USE_LIBPQXX
+
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <pqxx/pqxx>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_TYPE;
+}
+
+
+std::shared_ptr<NamesAndTypesList> fetchTableStructure(ConnectionPtr connection, const String & postgres_table_name, bool use_nulls)
+{
+    auto columns = NamesAndTypesList();
+
+    std::string query = fmt::format(
+           "SELECT attname AS name, format_type(atttypid, atttypmod) AS type, "
+           "attnotnull AS not_null, attndims AS dims "
+           "FROM pg_attribute "
+           "WHERE attrelid = '{}'::regclass "
+           "AND NOT attisdropped AND attnum > 0", postgres_table_name);
```

Comment:
Another concert about possible SQL injection.

Reply:
Seems ok because `'table_name'::regclass` will throw exception (which will now be caught with pqxx::undefined_table), if specified `table_name` is incorrect.